planner/cascades/implementation_rules.go

```diff
@@ -221,3 +224,22 @@ func (r *ImplSort) OnImplement(expr *memo.GroupExpr, reqProp *property.PhysicalP
 	)
 	return impl.NewSortImpl(ps), nil
 }
+
+// ImplHashAgg is the implementation rule which implements LogicalAggregation
+// to PhysicalHashAgg.
+type ImplHashAgg struct {
+}
+
+// Match implements ImplementationRule Match interface.
+func (r *ImplHashAgg) Match(expr *memo.GroupExpr, prop *property.PhysicalProperty) (matched bool) {
+	// TODO: deal with the hints when we have implemented StreamAgg.
+	return prop.IsEmpty()
+}
+
+// OnImplement implements ImplementationRule OnImplement interface.
+func (r *ImplHashAgg) OnImplement(expr *memo.GroupExpr, reqProp *property.PhysicalProperty) (memo.Implementation, error) {
+	la := expr.ExprNode.(*plannercore.LogicalAggregation)
+	hashAgg := plannercore.NewPhysicalHashAgg(la, reqProp.ExpectedCnt, &property.PhysicalProperty{ExpectedCnt: math.MaxFloat64})
```

Comment:
Seems like the expected count passed to children is always `math.MaxFloat64` for hash aggregation.

Reply:
Emmm, any problem of this?